### PR TITLE
Backup sidecar for postgres and mariadb *needs review*

### PR DESCRIPTION
**Pull request by Dexus**
creates a backup sidecar for postgres and mariadb.

ATTENTION: The values of postgres/mariadb needs a update to the last backup.image version - but we have current problems to upload it.

### CODE REVIEW

**Comment by fwiesel** [comment]
Can I convince you to use the hook `/docker-entrypoint-maintaindb.d/` for creating/maintaining the required users instead of having an own special script?

Any script (or sql file) in the  configmaps `db-maintain-<name>` will be executed on startup of the respective database.

**Comment by fwiesel** [comment]
Hi,

also it should not be located under `bin/`, but directly in that configmap, and end on `.sh`.
See here, how it will be [invoked](https://github.wdf.sap.corp/monsoon-docker/monsoon-postgres/blob/dba7c12f3ddab8e152f1fa7b475afaf63f89f45c/docker-entrypoint.sh#L103-L111).

File: postgres/values.yaml [hunk 1]
```diff
@@ -18,3 +20,6 @@ hba_cidr: 0.0.0.0/0
 persistence:
   enabled: true
 
+backup:
+  enabled: true
+  image: hub.global.cloud.sap/c5211168/monsoon3-backup:0.1.2
```
Comment (BugRoger) [inline postgres/values.yaml]: This needs to be configurable with a value. The default should be a public image `sapcc/backup-tools:v0.1.2`. Internally we overwrite this with out image from hub.global.cloud.sap.

File: mariadb/values.yaml [hunk 2]
```diff
@@ -7,6 +7,9 @@ name: DEFINED-IN-COMPONENT-CHART
 image: mariadb:10.1
 port_public: 3306
 #
-#root_password: 
+#root_password:
 #initdb_configmap:
 #persistance_claim:
+backup:
+  enabled: true
+  image: hub.global.cloud.sap/c5211168/monsoon3-backup:0.1.2
```
Comment (BugRoger) [inline mariadb/values.yaml]: This needs to be configurable with a value. The default should be a public image sapcc/backup-tools:v0.1.2. Internally we overwrite this with out image from hub.global.cloud.sap.

File: mariadb/templates/backup-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: mariadb-backup-{{.Values.name}}
+  labels:
+    system: openstack
+    type: configuration
+    component: database
+
+data:
+  mysql.backup.full: 3 days
+  mysql.backup.incr: 3 hours
```
Comment (BugRoger) [inline mariadb/templates/backup-configmap.yaml]: Can we make this configurable via Helm values? And defaults in the chart.

File: mariadb/values.yaml [hunk 4]
```diff
@@ -7,6 +7,9 @@ name: DEFINED-IN-COMPONENT-CHART
 image: mariadb:10.1
```
Comment (BugRoger) [inline mariadb/values.yaml]: make configurable

File: postgres/templates/backup-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+kind: ConfigMap
+metadata:
+  name: postgres-backup-{{.Values.name}}
+  labels:
+    system: openstack
+    type: configuration
+    component: database
+
+data:
+  postgres.backup.full: 3 days
```
Comment (BugRoger) [inline postgres/templates/backup-configmap.yaml]: make configurable

**Review by BugRoger — CHANGES_REQUESTED** [review]
make stuff more configurable

**Comment by Dexus** [comment]
Can you explain what is still needed to make the stuff more configurable? What am I currently missing?

**Comment by urfuwo** [comment]
@Dexus see the review above. 
* location and name of image need to be value. 
* backup scheduling behaviour need to be values.
* image need to be on sapcc with requested name
* image need to be published on hub.global interneally with same name
* build make file and docker container need to be published on sapcc/containers -> speak to Michael for setup.
* after merge: next concourse build for container with publishing to hub.docker and hub.global internally.

**Review by BugRoger — APPROVED** [review]

**Comment by fwiesel** [comment]
I see another problem with `postgres/templates/maintain-configmap.yaml`, it now conflicts with the individual maintain configmaps (Here an [example](https://github.com/sapcc/openstack-helm/blob/master/neutron/templates/db-maintain-configmap.yaml)).

In case of neutron (as the example given), you create now two `db-maintain-neutron` configmaps,
which in the best case will fail, or maybe simply end up applied in a semi-random order.

**Pull request closed without merging**